repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,193 @@
+        bufferedInputStream.reset()
+
+        // Wrapping the request to reset the inputStream/Reader flag
+        val wrappedRequest = new HttpServletRequestWrapper(request.asInstanceOf[HttpServletRequest], bufferedInputStream)
+        wrappedRequest.setAttribute("http://openrepose.org/queryParams", paramaterMap)
+
+        // Add the start time to be used by the ResponseTimeHandler/HttpLogFormatter.
+        wrappedRequest.setAttribute(START_TIME_ATTRIBUTE, startTime)
+
+        val scope = startSpan(wrappedRequest, tracer, ThisLogger, Tags.SPAN_KIND_CLIENT, reposeVersion, uriRedactionService)
+
+        // Conditionally remove the tracing header so it will be overwritten
+        val tracingHeaderConfig = reposeFilterLoader.getTracingHeaderConfig
+        if (tracingHeaderConfig.exists(_.isRewriteHeader)) wrappedRequest.removeHeader(TRACE_GUID)
+
+        //Grab the traceGUID from the request if there is one, else create one
+        val traceGUID = Option(wrappedRequest.getHeader(TRACE_GUID)) match {
+          case Some(string) if StringUtils.isNotBlank(string) => string
+          case _ => UUID.randomUUID.toString
+        }
+
+        MDC.put(TracingKey.TRACING_KEY, traceGUID)
+
+        try {
+          TryWith(filterContextList)(filterContextListToo => {
+            // Ensure the request URI is a valid URI
+            // This object is only being created to ensure its validity.
+            // So it is safe to suppress warning squid:S1848
+            new URI(wrappedRequest.getRequestURI)
+
+            if (tracingHeaderConfig.exists(_.isEnabled)) {
+              if (StringUtils.isBlank(wrappedRequest.getHeader(TRACE_GUID))) {
+                wrappedRequest.addHeader(TRACE_GUID, TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(VIA)))
+              }
+              if (tracingHeaderConfig.exists(_.isSecondaryPlainText)) {
+                TraceIdLogger.trace("Adding plain text trans id to request: {}", traceGUID)
+                wrappedRequest.replaceHeader(REQUEST_ID, traceGUID)
+              }
+              val tracingHeader = wrappedRequest.getHeader(TRACE_GUID)
+              TraceIdLogger.info("Tracing header: {}", TracingHeaderHelper.decode(tracingHeader))
+              wrappedResponse.addHeader(TRACE_GUID, tracingHeader)
+            }
+            new ReposeFilterChain(
+              filterContextListToo.filterContexts,
+              chain,
+              None,
+              optMetricRegistry,
+              tracer
+            ).doFilter(wrappedRequest, wrappedResponse)
+          }) match {
+            case Success(_) =>
+              ThisLogger.trace("{} -- Successfully processed request", nodeId)
+            case Failure(use: URISyntaxException) =>
+              ThisLogger.debug(s"$nodeId -- Invalid URI requested: ${wrappedRequest.getRequestURI}", use)
+              wrappedResponse.sendError(SC_BAD_REQUEST, "Error processing request")
```

Comment:
Having just said we don't have to worry about the response being mutable, all these send errors will fail if someone downstream committed the response.

Reply:
Yep. Updated to use the new `force` option of the wrapper's `sendError` method.